queue/priority_queue.go

```diff
@@ -42,50 +39,66 @@ type Item interface {
 
 type priorityItems []Item
 
+func (items *priorityItems) swap(i, j int) {
+	(*items)[i], (*items)[j] = (*items)[j], (*items)[i]
+}
+
+func (items *priorityItems) pop() Item {
```

Comment:
Oh, you're right, slice header mutation requires a pointer receiver. Sorry, my mistake!